src/runtime-rs/crates/runtimes/virt_container/src/container_manager/container.rs

```diff
@@ -0,0 +1,405 @@
+        let logger = logger_with_process(container_process);
+        info!(logger, "start wait process");
+
+        let inner = self.inner.read().await;
+        inner
+            .fetch_exit_watcher(container_process)
+            .context("fetch exit watcher")
+    }
+
+    pub async fn kill_process(
+        &self,
+        container_process: &ContainerProcess,
+        signal: u32,
+        all: bool,
+    ) -> Result<()> {
+        let inner = self.inner.read().await;
+        inner.signal_process(container_process, signal, all).await
+    }
+
+    pub async fn exec_process(
+        &self,
+        container_process: &ContainerProcess,
+        stdin: Option<String>,
+        stdout: Option<String>,
+        stderr: Option<String>,
+        terminal: bool,
+        oci_process: OCIProcess,
+    ) -> Result<()> {
+        let process = Process::new(
+            container_process,
+            self.pid,
+            &self.config.bundle,
+            stdin,
+            stdout,
+            stderr,
+            terminal,
+        );
+        let exec = Exec {
+            process,
+            oci_process,
+        };
+        let mut inner = self.inner.write().await;
+        inner.add_exec_process(&container_process.exec_id, exec);
+        Ok(())
+    }
+
+    pub async fn close_io(&self, container_process: &ContainerProcess) -> Result<()> {
+        let mut inner = self.inner.write().await;
+        inner.close_io(container_process).await
+    }
+
+    pub async fn stop_process(&self, container_process: &ContainerProcess) -> Result<()> {
+        let mut inner = self.inner.write().await;
+        inner
+            .stop_process(container_process, true)
+            .await
+            .context("stop process")
+    }
+
+    pub async fn pause(&self) -> Result<()> {
+        let inner = self.inner.read().await;
+        if inner.init_process.status == ProcessStatus::Paused {
+            warn!(self.logger, "container is paused no need to pause");
+            return Ok(());
+        }
+        self.agent
+            .pause_container(self.container_id.clone().into())
+            .await
+            .context("agent pause container")?;
+        Ok(())
+    }
+
+    pub async fn resume(&self) -> Result<()> {
+        let inner = self.inner.read().await;
+        if inner.init_process.status == ProcessStatus::Running {
+            warn!(self.logger, "container is running no need to resume");
+            return Ok(());
+        }
+        self.agent
+            .resume_container(self.container_id.clone().into())
+            .await
+            .context("agent pause container")?;
+        Ok(())
+    }
+
+    pub async fn resize_pty(
+        &self,
+        process: &ContainerProcess,
+        width: u32,
+        height: u32,
+    ) -> Result<()> {
+        let logger = logger_with_process(process);
+        let inner = self.inner.read().await;
+        if inner.init_process.status == ProcessStatus::Running {
```

Comment:
Copy miss?

Reply:
Thanks for your review, it's trying to check process not running, change to `inner.init_process.status != ProcessStatus::Running`.